packages/vite-environment-provider-workerd/src/moduleUtils.ts

```diff
@@ -0,0 +1,71 @@
+import { init as initCjsModuleLexer, parse } from 'cjs-module-lexer';
+import { dirname, resolve } from 'node:path';
+import { readFile, stat } from 'node:fs/promises';
+
+export async function collectModuleInfo(
+  moduleCode: string,
+  moduleFilePath: string,
+): Promise<CjsModuleInfo | EsmModuleInfo> {
+  if (!isCommonJS(moduleCode)) {
+    return { isCommonJS: false };
+  }
+
+  await initCjsModuleLexer();
+
+  const namedExportsSet = new Set<string>();
+
+  const cjsLexerResult = parse(moduleCode);
+  cjsLexerResult.exports.forEach(namedExport =>
+    namedExportsSet.add(namedExport),
+  );
```

Comment:
Can we use a loop rather than `forEach` for consistency (makes it a little easier to scan).

```suggestion
  for (const namedExport of cjsLexerResult.exports) {
    namedExportsSet.add(namedExport)
  }
```